src/dra/loading.c

```diff
@@ -735,16 +734,16 @@ void HandleEnding(void) {
             if (LoadFileSim(4, SimFileType_System) < 0) {
                 break;
             }
-            LoadClut(g_Clut, 0x200, 0xF0);
         }
+        LoadClut((u_long*)g_Clut, 0x200, 0xF0);
         g_GameStep++;
         break;
     case 2:
         if (g_UseDisk) {
             g_CdStep = CdStep_LoadInit;
             g_LoadFile = CdFile_StagePrg;
         }
-        g_GameStep = 3;
+        g_GameStep++;
```

Comment:
This is an interesting discovery! We did not know GCC would optimise the `++` in such way

Reply:
Yep, got this from [the PSP side](https://decomp.me/scratch/9QZjQ) which I was using to try to work out some mismatches (though I did not fully solve it there)